tests/airport.spec.js

```diff
@@ -0,0 +1,450 @@
+
+// Clean Up
+expected = undefined;
+actual = undefined;
+result = undefined;
+airport = null;
+plane = null;
+
+// Test 15 - prevent takeOff when the weather is stormy 
+console.log(`==========`);
+console.log(`Test 15 - prevent takeOff when the weather is stormy`);
+
+// Arrange
+var testWeather = { isItStormy: () => true };
+airport = new Airport(testWeather);
+plane = new Plane(`test1`);
+airport.planesLanded = [plane];
+expected = 1;
+
+// Act
+airport.takeOff(plane);
+actual = airport.planesLanded.length;
+
+// Assert
+result = assertEquals(expected, actual);
+console.log(`Test 15 - prevent takeOff when the weather is stormy: ${result}`);
+
+// Clean Up
+expected = undefined;
+actual = undefined;
+result = undefined;
+airport = null;
+plane = null;
+testWeather = undefined;
+
+// Test 16 - prevent landing when the weather is not stormy
+console.log(`==========`);
+console.log(`Test 16 - prevent landing when the weather is not stormy`);
+
+// Arrange
+testWeather = { isItStormy: () => true };
+airport = new Airport(testWeather);
+plane = new Plane(`test1`);
+expected = 0;
+
+// Act
+airport.landPlane(plane);
+actual = airport.planesLanded.length;
+
+// Assert
+result = assertEquals(expected, actual);
+console.log(`Test 16 - prevent landing when the weather is not stormy: ${result}`);
+
+// Clean Up
+expected = undefined;
+actual = undefined;
+result = undefined;
+airport = null;
+plane = null;
+testWeather = undefined;
+
+// Test 17 - planes that have landed must be at an airport
+// The only way for planes to land is via an airport, so this adds functionality to ask a plane which airport it is at.
+console.log(`==========`);
+console.log(`Test 17 - planes that have landed must be at an airport`);
+
+// Arrange
+airport = new Airport(undefined, `testairport`);
+plane = new Plane(`test1`);
+expected = 'testairport';
+
+// Act
+airport.landPlane(plane);
+actual = plane.getAirport();
+
+// Assert
+result = assertEquals(expected, actual);
+console.log(`Test 17 - planes that have landed must be at an airport: ${result}`);
+
+// Clean Up
+expected = undefined;
+actual = undefined;
+result = undefined;
+airport = null;
+plane = null;
+testWeather = undefined;
+
+// Test 18 - planes that have taken off returns 'In the air' when getAirport is called
+console.log(`==========`);
+console.log(`Test 18 - planes that have taken off returns 'In the air' when getAirport is called`);
+
+// Arrange
+airport = new Airport(undefined, `testairport`);
+plane = new Plane(`test1`);
+airport.landPlane(plane);
+expected = 'In the air';
+
+// Act
+airport.takeOff(plane);
+actual = plane.getAirport();
+
+// Assert
+result = assertEquals(expected, actual);
+console.log(`Test 18 - planes that have taken off returns 'In the air' when getAirport is called: ${result}`);
+
+// Clean Up
+expected = undefined;
+actual = undefined;
+result = undefined;
+airport = null;
+plane = null;
+testWeather = undefined;
```

Comment:
I felt like there were some aspects I didn't test properly at the end (e.g. Weather class, Plane class)